identity_storage/src/stronghold_storage.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_sdk::client::secret::stronghold::StrongholdSecretManager;
+use iota_sdk::client::secret::SecretManager;
+use iota_stronghold::Stronghold;
+use std::sync::Arc;
+use tokio::sync::MutexGuard;
+
+/// Wrapper around `StrongholdSecretManager` that implements the [`KeyIdStorage`](crate::KeyIdStorage)
+/// and [`JwkStorage`](crate::JwkStorage) interfaces.
+#[derive(Clone, Debug)]
+pub struct StrongholdStorage(Arc<SecretManager>);
+
+impl StrongholdStorage {
+  /// Creates a new [`StrongholdStorage`].
+  pub fn new(stronghold_secret_manager: StrongholdSecretManager) -> Self {
+    Self(Arc::new(SecretManager::Stronghold(stronghold_secret_manager)))
+  }
+
+  /// Shared reference to the inner [`SecretManager`].
+  pub fn as_secret_manager(&self) -> Arc<SecretManager> {
```

Comment:
```suggestion
  pub fn as_secret_manager(&self) -> &SecretManager {
```

Reply:
This doesn't work, `cannot return reference to temporary value ...`

Reply:
```
  pub fn as_secret_manager(&self) -> &SecretManager {
    self.0.as_ref()
  }
```

Replacing the function with this compiles fine for me.

Reply:
thanks, my bad, I replaced it